src/methods/wrapExceptions.ts

```diff
@@ -19,13 +19,15 @@ export const wrapExceptions = async <T = unknown>(
   }
 };
 
+export interface QueryState<T, E> {
```

Comment:
Just curious could you explain how this should be used? Is this what you were saying is intended to replace try catch?